packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -252,3 +256,62 @@ class Variable extends TreeNode<Variable> {
     return '${boundVar.name} - $value';
   }
 }
+
+/// A node in a tree of scripts.
+///
+/// A node can either be a directory (a name with potentially some child nodes),
+/// a script reference (where [scriptRef] is non-null), or a combination of both
+/// (where the node has a non-null [scriptRef] but also contains child nodes).
+class FileNode extends TreeNode<FileNode> {
+  FileNode(this.name);
+
+  final String name;
+
+  // This can be null.
+  ScriptRef scriptRef;
+
+  /// This exists to allow for O(1) lookup of children when building the tree.
+  Map<String, FileNode> _childrenAsMap = {};
```

Comment:
in the spirit of NNBD can you make this final and clear the entries rather than setting it to null?

Reply:
Hmm, we'd save some of the memory then but not all (the n instances of empty maps). It could very well be that this is an unnecessary optimization (the `_trimChildrenAsMapEntries()` method generally).

I'll switch to clearing the map -

Reply:
Converted to `_childrenAsMap.clear()`.